### PR TITLE
feat: introducing a new example (client and server code)

### PR DESCRIPTION
**Pull request by cr0wg4n**
Hi @psibean. I added one example more. It is in the "complete" folder, and I moved your example to the "simple" folder. I hope this helps to some people who wants to use your library.

### CODE REVIEW

**Comment by psibean** [comment]
Ah what. The fail here is my bad.

I thought I had removed the Run Linters job from the workflow (as I made it a step).

I had some weird funky branching issues when I was testing it all out, so I must not have had the removal merged. 😅

**Comment by psibean** [comment]
Thanks for this @cr0wg4n !

If you update the dotenv import to a command line require, perhaps a small README would be helpful too?

I'm in the process of making an example express app with email/password authentication (with a one time code required for registration), PKCE OAuth2 flow (discord), and thus session based with for server side state.

I plan on using csrf-csrf with session unique secrets, and so hopefully that will help provide an "advanced" example too!

**Comment by cr0wg4n** [comment]
Done, it's a pleasure. 

Sounds good, nice work! ;D. I'll be back haha

**Comment by psibean** [comment]
So not your problem, but to fix the linting here, we likely need t do the following:

Update the existing eslint config with:

```
  parserOptions: {
    tsconfigRootDir: __dirname,
    project: ['./tsconfig.lint.json'],
  },
```

And then create a new `tsconfig.ling.json` like so:

```
{
  "extends": "./tsconfig.json",
  "include": ["./src/**/*", "./example/**/*"]
}
```

**Comment by psibean** [comment]
> Done, it's a pleasure.
> 
> Sounds good, nice work! ;D. I'll be back haha

Wait, I'm confused.

You deleted the config file? All you needed to do was remove the dotenv import, and then update the script in package.json to:

```
"start": "node -r dotenv/config ./src/index.js"
```

You can still use dotenv, just don't import it, import it on the command line instead!

**Comment by cr0wg4n** [comment]
> So not your problem, but to fix the linting here, we likely need to do the following:
> 
> Update the existing eslint config with:
> 
> ```
>   parserOptions: {
>     tsconfigRootDir: __dirname,
>     project: ['./tsconfig.lint.json'],
>   },
> ```
> 
> And then create a new `tsconfig.lint.json` like so:
> 
> ```
> {
>   "extends": "./tsconfig.json",
>   "include": ["./src/**/*", "./example/**/*"]
> }
> ```

This change lints the example subfolders with js files as ts.

**Comment by cr0wg4n** [comment]
> > Done, it's a pleasure.
> > Sounds good, nice work! ;D. I'll be back haha
> 
> Wait, I'm confused.
> 
> You deleted the config file? All you needed to do was remove the dotenv import, and then update the script in package.json to:
> 
> ```
> "start": "node -r dotenv/config ./src/index.js"
> ```
> 
> You can still use dotenv, just don't import it, import it on the command line instead! This will load the `.env` environment variables without importing the dependency. This decouples the runtime from dotenv.

Sorry for the confusion

File: example/complete/src/config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import 'dotenv/config';
```
[the remark this answers is not in the document]
Comment (cr0wg4n) [reply example/complete/src/config.js]: Okay, thanks for the advice

**Comment by psibean** [comment]
> This change lints the example subfolders with js files as ts.

Ah yeah, that's a pain.

So I think we would need something like this in the eslint config:

```
{
    extends: [
      'eslint:recommended'
    ],
    root: true,
    ignorePatterns: [
      ".eslintrc.cjs",
      "lib/**/*",
      "exanple/**/*",
      "example/src/**/*",
    ],
    overrides: [
      {
        files: ["**/*.ts", "**/*.tsx"],
        extends: [
          "eslint:recommended",
          "plugin:@typescript-eslint/eslint-recommended",
          "plugin:@typescript-eslint/recommended"
        ],
        parser: "@typescript-eslint/parser",
          parserOptions: {
            tsconfigRootDir: __dirname,
            project: ["./tsconfig.json"],
          },
        plugins: ["@typescript-eslint"],
      }
    ]
  };
```

**Comment by psibean** [comment]
Hope you don't mind, I fixed the linting on the branch on your fork and merged it in.

**Comment by cr0wg4n** [comment]
thank you :D. Great

**Pull request merged**